cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go

```diff
@@ -264,21 +270,33 @@ func parseIntKey(annotations map[string]string, key string) (resource.Quantity,
 }
 
 func parseCPUCapacity(annotations map[string]string) (resource.Quantity, error) {
-	return parseKey(annotations, cpuKey)
+	// Checking both sets of annotations. If the upstream annotation doesn't exist,
+	// get the value of the deprecated one. If neither exist, return nil.
+	if val, err := parseKey(annotations, cpuKey); val != zeroQuantity && err == nil {
+		return val, err
+	} else if err != nil {
+		return zeroQuantity.DeepCopy(), err
+	}
+	// If we don't find the old annotation, parseIntKey returns zeroQuantity.
+	return parseKey(annotations, deprecatedCpuKey)
 }
 
 func parseMemoryCapacity(annotations map[string]string) (resource.Quantity, error) {
-	// The value for the memoryKey is expected to be an integer representing Mebibytes. e.g. "1024".
+	// The value for the deprecatedMemoryKey is expected to be an integer representing Mebibytes. e.g. "1024".
```

Comment:
i like the comments you have on the other functions about "Checking both sets of annotations...", would be nice to have something similar here as well.

Reply:
Added some more comments :)